File: frontend/lib/models/auth/login_response.dart
import 'user_role.dart';

/// DTO representing the response payload from POST /api/auth/login.
class LoginResponse {
  final String token;
  final String userId;
  final String name;
  final UserRole role;

  const LoginResponse({
    required this.token,
    required this.userId,
    required this.name,
    required this.role,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      token: json['token'] as String,
      userId: json['userId'] as String,
      name: json['name'] as String,
      role: UserRole.fromString(json['role'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'token': token,
      'userId': userId,
      'name': name,
      'role': role.toBackendString(),
    };
  }
}

File: backend/src/main/java/com/doorworkflow/dto/response/LoginResponse.java
package com.doorworkflow.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

/**
 * DTO for login response.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class LoginResponse {
    private String token;
    private UUID userId;
    private String name;
    private String role;
}
